src/Microsoft.SqlTools.ServiceLayer/Workspace/Workspace.cs

```diff
@@ -142,17 +153,41 @@ private string ResolveFilePath(string filePath)
             return filePath;
         }
 
-        internal static bool IsPathInMemory(string filePath)
+        internal static bool IsPathInMemoryOrNonFileUri(string path)
         {
-            // When viewing SqlTools files in the Git diff viewer, VS Code
-            // sends the contents of the file at HEAD with a URI that starts
-            // with 'inmemory'.  Untitled files which have been marked of
-            // type SqlTools have a path starting with 'untitled'.
-            return
-                filePath.StartsWith("inmemory:") ||
-                filePath.StartsWith("tsqloutput:") ||
-                filePath.StartsWith("git:") ||            
-                filePath.StartsWith("untitled:");
+            try
+            {
+                string scheme = GetScheme(path);
+                if (scheme != null && scheme.Length > 0 && !scheme.Equals("file"))
+                {
+                    return true;
+                }
+            }
+            catch(Exception)
+            {
+                // Intentionally skipping if there is a parse error
+            }
+
+            return false;
+        }
+
+        public static string GetScheme(string uri)
```

Comment:
just a suggestion ... these new methods are all for parsing a file path and can be all moved to a new class so the tests can only focus those methods.